yt_dlp/extractor/camfm.py

```diff
@@ -0,0 +1,87 @@
+from re import finditer
+from urllib.parse import urljoin
+
+from .common import InfoExtractor
+from ..utils import join_nonempty, unified_timestamp
+
+
+class CamFMBaseIE(InfoExtractor):
+    def _absolute_url(self, path, root='https://camfm.co.uk'):
+        return urljoin(root, path)
+
+
+class CamFMShowIE(CamFMBaseIE):
+    _VALID_URL = r'https://(?:www\.)?camfm\.co\.uk/shows/(?P<id>[^/]+)/?'
+    _TESTS = [{
+        'playlist_mincount': 0,
+        'url': 'https://camfm.co.uk/shows/soul-mining/',
+        'info_dict': {
+            'id': 'soul-mining',
+            'thumbnail': 'md5:6a873091f92c936f23bdcce80f75e66a',
+            'title': 'Soul Mining',
+            'description': 'Telling the stories of jazz, funk and soul from all corners of the world.',
+        },
+    }]
+
+    def _real_extract(self, url):
+        show_id = self._match_id(url)
+        page = self._download_webpage(url, show_id)
+        episodes = finditer(r"javascript:popup\('(/player/[^']+)', 'listen'", page)
+        urls = [self._absolute_url(m.group(1)) for m in episodes]
+
+        return {
+            '_type': 'playlist',
+            'id': show_id,
+            'entries': [self.url_result(i, CamFMEpisodeIE) for i in urls],
+            'thumbnail': self._absolute_url(self._search_regex(
+                r'<img[^>]+class="thumb-expand"[^>]+src="([^"]+)"', page, 'thumbnail', fatal=False)),
+            'title': self._html_search_regex('<h1>([^<]+)</h1>', page, 'title', fatal=False),
+            'description': self._html_search_regex('<div class="small-12 medium-8 cell">\n(.*)',
```

Comment:
`.*` is a bad idea. Can you refine it?

Reply:
or `get_element_by_class` might be useful for the description (along with `clean_html` -- both are `utils`)